10-Reflection/Reflection.Tasks/CommonTasks.cs

```diff
@@ -59,9 +83,23 @@ public static class CommonTasks
         /// <param name="obj">source object to set property to</param>
         /// <param name="propertyPath">dot-separated property path</param>
         /// <param name="value">assigned value</param>
-        public static void SetPropertyValue(this object obj, string propertyPath, object value) {
-            // TODO : Implement SetPropertyValue method
-            throw new NotImplementedException();
+        public static void SetPropertyValue(this object obj, string propertyPath, object value)
+        {
+            string[] parts = propertyPath.Split('.');
+            string path = propertyPath;
+            object root = obj;
+
+            if (parts.Length > 1)
+            {
+                path = parts[parts.Length - 1];
+                parts = parts.TakeWhile((p, i) => i < parts.Length - 1).ToArray();
+
+                string path2 = String.Join(".",parts);
+                root = obj.GetPropertyValue<object>(path2);
+            }
+
+            var property = root.GetType().BaseType.GetProperty(path);
```

Comment:
https://allisonmaruska.files.wordpress.com/2015/10/we-need-to-go-deeper.jpg

Reply:
works fine with more property depth :)
either using recursion here a bad practice?)
just add some test with additional property for Manager class

Reply:
recursion is certainly a bad practice and can be easily avoided here, but i meant deeper in class hierarchy